src/views/author_notes.ts

```diff
@@ -14,14 +14,101 @@
  * limitations under the License.
  */
 
-import { BaseNodeProps, BaseNodeView } from './base_node_view'
+import { ManuscriptNode, schema } from '@manuscripts/transform'
+
+import BlockView from './block_view'
 import { createNodeView } from './creators'
+import { EditableBlockProps } from './editable_block'
 
-export class AuthorNotesView extends BaseNodeView<BaseNodeProps> {
+export class AuthorNotesView extends BlockView<EditableBlockProps> {
   container: HTMLElement
+  allowedTypes = ['corresp', 'footnote', 'paragraph']
+  headersStrings: { [key: string]: string } = {
+    corresp: 'Correspondence',
+    footnote: 'Author notes',
+    paragraph: 'Author notes',
+  }
 
   public ignoreMutation = () => true
   public stopEvent = () => true
+
+  public createElement = () => {
+    this.container = document.createElement('div')
+    this.container.classList.add('block', 'author-notes-container')
+    this.dom.appendChild(this.container)
+  }
+
+  public updateContents = () => {
+    this.dom.setAttribute('contenteditable', 'false')
+    this.container.innerHTML = ''
+
+    const items: { [key: string]: ManuscriptNode[] } = {}
+    this.allowedTypes.forEach((type) => (items[type] = []))
+
+    this.view.state.doc.descendants((node) => {
+      if (node.type === schema.nodes.author_notes) {
+        node.descendants((anNode) => {
+          if (this.allowedTypes.includes(anNode.type.name)) {
+            items[anNode.type.name].push(anNode)
+            return false
+          }
+        })
+        return false
+      }
+    })
+
+    this.allowedTypes.forEach((type) => {
+      if (items[type] && items[type].length) {
+        const showHeader =
+          type === this.allowedTypes[0] ||
+          (type === this.allowedTypes[1] &&
+            (!!items[this.allowedTypes[1]]?.length ||
+              !!items[this.allowedTypes[2]]?.length)) ||
+          (type === this.allowedTypes[2] &&
+            !items[this.allowedTypes[1]]?.length &&
+            !!items[this.allowedTypes[2]]?.length)
```

Comment:
if all allowedTypes should have a header, then you can have showHeader always true, instead of doing these logical operations, because you are already checking if its allowed or not in line 61

Reply:
It is completely rewritten